cookbooks/cdo-varnish/test/integration/cloudfront/minitest/test_default.rb

```diff
@@ -0,0 +1,315 @@
+PID = spawn('cd ~; java -jar mock.jar')
+# Don't start tests until wiremock is live.
+mock_started = false
+until mock_started
+  `sleep 1`
+  puts 'testing mock'
+  `curl -s -i #{LOCALHOST}:#{DASHBOARD_PORT}/start`
+  mock_started = $?.exitstatus == 0
+end
+puts 'setup finished'
+
+at_exit do
+  puts 'teardown'
+  `curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/shutdown`
+  Process.wait PID
+  `kill -INT #{NGROK_PID}`
+  Process.wait NGROK_PID
+  puts 'teardown finished'
+end
+
+require 'json'
+require 'minitest/autorun'
+
+describe 'varnish' do
+  it 'is installed' do
+    assert_equal '/usr/sbin/varnishd', `which varnishd`.strip
+  end
+
+  it 'is running' do
+    assert_equal 0, (`service varnish status` && $?.exitstatus)
+  end
+end
+
+# Returns the number of times the specified stub-URL was accessed.
+def count_requests(url)
+  request = <<JSON
+{
+  "method": "GET",
+  "url": "#{url}"
+}
+JSON
+  JSON.parse(`curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/requests/count -d '#{request}'`)['count']
+end
+
+# Mocks a simple text/plain response body at the specified URL.
+def mock_url(url, body, request_headers={}, response_headers={}, method='GET')
+  json = {
+  request: {
+    method: method,
+    url: url,
+    headers: Hash[*request_headers.map{|k, v|[k,{equalTo: v}]}.flatten]
+  },
+  response: {
+    status: 200,
+    body: body,
+    headers: {
+      'Cache-Control' => 'public, max-age=10',
+      'Content-Type' => 'text/plain'
+    }.merge(response_headers)
+  }
+}.to_json
+  `curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/mappings/new -d '#{json}'`
+end
+
+def _request(url, headers={}, cookies={}, method='GET')
+  header_string = headers.map { |key, value| "-H \"#{key}: #{value}\"" }.join(' ')
+  cookie_string = cookies.empty? ? '' : "--cookie \"#{cookies.map{|k,v|"#{k}=#{v}"}.join(';')}\""
+  `curl -X #{method} -s #{cookie_string} #{header_string} -i #{url}`.tap{assert_equal 0, $?.exitstatus}
+end
+
+def request(url, headers={}, cookies={})
+  _request("#{LOCALHOST}:#{DASHBOARD_PORT}#{url}", headers, cookies)
+end
+
+def proxy_request(url, headers={}, cookies={}, method='GET')
+  headers.merge!(host: 'cdo.ngrok.io')
+  headers.merge!('X-Forwarded-Proto' => 'https')
+  _request("#{CLOUDFRONT_URL}#{url}", headers, cookies, method)
+end
+
+def code(response)
+  /HTTP\/1\.1 (\d+)/.match(response.lines.to_a.first.strip)[1].to_i
+end
+
+def assert_ok(response)
+  assert_equal 200, code(response)
+end
+def assert_cache(response, hit)
+  assert_equal hit ? 'Hit' : 'Miss', /X-Cache: (\w+) from cloudfront/.match(response)[1]
+end
+def assert_miss(response)
+  assert_cache response, false
+end
+def assert_hit(response)
+  assert_cache response, true
+end
+
+def last_line(response)
+  response.lines.to_a.last.strip
+end
+
+require 'securerandom'
+def get_url(prefix='x', suffix='')
+  id = SecureRandom.uuid
+  '/' + prefix.to_s + '/' + id + '/' + suffix.to_s
+end
+
+describe 'http' do
```

Comment:
I think we are are really describing here is say "http_proxy_caching" rather than just "http", might be clearer if you changed it to that so they know what "it" means below.